Dockerfile

```diff
@@ -16,10 +16,21 @@ RUN cd ./backend && go build -o ./server ./cmd/
 
 RUN cd ./frontend && npm install && npm run build
 
-# Create a plugins folder if none exists so we copy whatever exists to the
-# image.
+# Backwards compatibility, move plugin folder to only copy matching plugins.
+RUN mv plugins plugins-old
+
 RUN mkdir -p ./plugins
 
+# Backwards compatibility, copy any plugins found inside "./plugins" into ".plugins".
+# They should match plugins/MyFolder/main.js, otherwise they are not copied.
+RUN find plugins-old -maxdepth 2 -type f  | grep -i main.js$ | xargs -i dirname {} | xargs -i cp -a {} ./plugins/
+
+RUN find .plugins -maxdepth 2 -type f  | grep -i main.js$ | xargs -i dirname {} | xargs -i cp -a {} ./plugins/
+
+# Backwards compatibility, plugins could also be in a MyPlugin/dist/main.js file.
+RUN for i in $(find ./plugins-old/*/dist/main.js); do plugin_name=$(echo $i|cut -d'/' -f3); mkdir -p plugins/$plugin_name; cp $i plugins/$plugin_name; done
```

Comment:
I would run this before running the new way (the .plugins), so any new way would overwrite a forgotten old format.
Either that or fail if the path exists when copying (basically to prevent overwriting plugins from  an existing old format, but running before should be enough).

Reply:
I moved this to before.